core/src/main/java/io/stargate/core/BundleUtils.java

```diff
@@ -0,0 +1,21 @@
+package io.stargate.core;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class BundleUtils {
+  private static final Logger logger = LoggerFactory.getLogger(CoreActivator.class);
+
+  /** Gets the instance of the newly registered service. */
+  public static Object getRegisteredService(BundleContext context, ServiceEvent serviceEvent) {
```

Comment:
Could we make it `public static <T> T getRegisteredService(BundleContext context, ServiceEvent serviceEvent, Class<T> serviceType)` to avoid type checks at calling sites?

Also, if the caller does not like the object's type it will have to "unget" the service to avoid dangling references (I think OSGi requires that).

Reply:
tbh, I extracted the method from `persistence-cassandra-3.11`, `restapi`, ...

Good point, I've modified the utility method. Is that what you had in mind?